src/Stratis.Bitcoin.Features.BlockStore/Models/BlockTransactionDetailsModel.cs

```diff
@@ -13,7 +13,11 @@ public class BlockTransactionDetailsModel : BlockModel
 
         public BlockTransactionDetailsModel(Block block, Network network, ChainBase chain) : base(block, chain)
         {
-            this.Transactions = block.Transactions.Select(trx => new TransactionVerboseModel(trx, network)).ToArray();
+            this.Transactions = new TransactionVerboseModel[block.Transactions.Count];
```

Comment:
@maciejzaleski please revert this as this is called so infrequently that there is no benefit to change this to a `for` statement. It will also make the PR smaller. We don't need to change to `for`s everywhere we use a select, it needs to be on merit and readability needs to be considered at times.